src/index.spec.ts

```diff
@@ -4,9 +4,149 @@ import {
+    expect(() => removeDuplicatesFromArray(null)).toThrowError(errorMessage);
+    expect(() => removeDuplicatesFromArray(undefined)).toThrowError(errorMessage);
+    expect(() => removeDuplicatesFromArray({} as any)).toThrowError(errorMessage);
+  });
+  it('should return an array with no duplicates', () => {
+    const oneElement = [1];
+    expect(removeDuplicatesFromArray(oneElement)).toEqual([1]);
+
+    const numbers = [1, 2, 3, 4, 5, 6];
+    expect(removeDuplicatesFromArray(numbers)).toEqual([1, 2, 3, 4, 5, 6]);
+
+    const repeatedNumbers = [1, 1, 1, 1, 1, 1];
+    expect(removeDuplicatesFromArray(repeatedNumbers)).toEqual([1]);
+
+    const strings = ['a', 'b', 'c', 'd', 'e', 'f'];
+    expect(removeDuplicatesFromArray(strings)).toEqual(['a', 'b', 'c', 'd', 'e', 'f']);
+
+    const repeatedStrings = ['a', 'a', 'a', 'a', 'a', 'a'];
+    expect(removeDuplicatesFromArray(repeatedStrings)).toEqual(['a']);
+
+    const stringAndNumbers = [1, 'b', 'c', 4, 'e', 6];
+    expect(removeDuplicatesFromArray(stringAndNumbers)).toEqual([1, 'b', 'c', 4, 'e', 6]);
+
+    const repeatedStringAndNumbers = ['a', 'a', 1, 1, null, undefined];
+    expect(removeDuplicatesFromArray(repeatedStringAndNumbers)).toEqual(['a', 1, null, undefined]);
+  })
 });
+
+import products from './utils/products';
+
+describe('createProduct()', () => {
+  it('should throw an error if product is not valid', () => {
+    for(const product of products){
+      expect(() => createProduct(product)).toThrowError();
+    }
+  });
+  it('should return a product', () => {
+    const validProduct = {  
+      name: 'Egg',
+      tags: ['dairy'],
+      description: 'A boiled egg',
+      price: 2.30,
+    };
+    expect(createProduct(validProduct)).toEqual({
+      id: expect.any(Number),
+      ...validProduct,
+    });
+  });
+});
+
+import users from './utils/users';
+
+describe('createRandomProduct()', () => {
+  it('should throw an error if user is not creator', () => {
+    const defaultMessage = 'You are not allowed to create products';
+    expect(() => createRandomProduct(users[0].email)).toThrowError(defaultMessage);
+    expect(() => createRandomProduct(users[2].email)).toThrowError(defaultMessage);
+  });
+  it('should throw an error if email is incorrect', () => {
+    expect(() => createRandomProduct(null)).toThrowError();
+    expect(() => createRandomProduct(undefined)).toThrowError();
+    expect(() => createRandomProduct('')).toThrowError();
+  });
+  it('should return a product when user is creator', () => {
+    expect(createRandomProduct(users[1].email)).toEqual({
+      id: expect.any(Number),
+      name: expect.any(String),
+      description: expect.any(String),
+      tags: expect.any(Array),
+      price: expect.any(String),
+    })
+  });
+});
+
+import { mocked } from 'ts-jest/utils';
+import fetch from 'node-fetch';
+
+jest.mock('node-fetch', () => {
+  return jest.fn();
+});
+
+beforeEach(() => {
+  mocked(fetch).mockClear();
+});
```

Comment:
Commonly, arrange general logic like this is located before any test. I mean before line 14.

Now, if the `beforeEach()` block is needed only for testing `getStarWarsPlanets`, you can move it inside the `describe('getStarWarsPlanets')` block.